app/models/dunning_campaign.rb

```diff
@@ -10,7 +10,8 @@ class DunningCampaign < ApplicationRecord
   belongs_to :organization
 
   has_many :thresholds, class_name: "DunningCampaignThreshold", dependent: :destroy
-  has_many :customers, foreign_key: :applied_dunning_campaign_id, dependent: :nullify
```

Comment:
We do not destroy dunning_campaigns - we discard them, so this nullify never works